src/handlers/decision.rs

```diff
@@ -0,0 +1,340 @@
+use crate::github;
+use crate::{
+    config::DecisionConfig, db::issue_decision_state::*, github::Event, handlers::Context,
+    interactions::ErrorComment,
+};
+use anyhow::bail;
+use anyhow::Context as Ctx;
+use chrono::{DateTime, Duration, Utc};
+use parser::command::decision::Resolution;
+use parser::command::decision::*;
+use serde::{Deserialize, Serialize};
+use std::collections::BTreeMap;
+
+pub const DECISION_PROCESS_JOB_NAME: &str = "decision_process_action";
+
+#[derive(Serialize, Deserialize)]
+pub struct DecisionProcessActionMetadata {
+    pub message: String,
+    pub get_issue_url: String,
+    pub status: Resolution,
+}
+
+pub(super) async fn handle_command(
+    ctx: &Context,
+    _config: &DecisionConfig,
+    event: &Event,
+    cmd: DecisionCommand,
+) -> anyhow::Result<()> {
+    let db = ctx.db.get().await;
+
+    let DecisionCommand {
+        resolution,
+        reversibility,
+        team: team_name,
+    } = cmd;
+
+    let issue = event.issue().unwrap();
+    let user = event.user();
+
+    match get_issue_decision_state(&db, &issue.number).await {
+        Ok(_state) => {
+            // TO DO
+            let cmnt = ErrorComment::new(
+                &issue,
+                "We don't support having more than one vote yet. Coming soon :)",
+            );
+            cmnt.post(&ctx.github).await?;
+
+            Ok(())
+        }
+        _ => {
+            let is_team_member = user.is_team_member(&ctx.github).await.unwrap_or(false);
+            if !is_team_member {
+                let cmnt = ErrorComment::new(
+                    &issue,
+                    "Only team members can be part of the decision process.",
+                );
+                cmnt.post(&ctx.github).await?;
+
+                return Ok(());
+            }
+
+            match team_name {
+                None => {
+                    let cmnt = ErrorComment::new(
+                        &issue,
+                        "In the first vote, is necessary to specify the team name that will be involved in the decision process.",
+                    );
+                    cmnt.post(&ctx.github).await?;
+
+                    Ok(())
+                }
+                Some(team_name) => {
+                    match github::get_team(&ctx.github, &team_name).await {
+                        Ok(Some(team)) => {
+                            let start_date: DateTime<Utc> = chrono::Utc::now().into();
+                            let end_date: DateTime<Utc> =
+                                start_date.checked_add_signed(Duration::days(10)).unwrap();
```

Comment:
Just FYI - the 10 days are typically after consensus is reached, i.e., they don't start immediately.

> The time-delay for feedback starts only when the necessary threshold for approval is otherwise met, including there not being any raised objections. If objections are raised and resolved during the time-delay, the waiting period starts again.

Reply:
Starts immediately for `reversible` decisions but not for `irreversible` decisions: https://lang-team.rust-lang.org/decision_process/examples.html#stabilizing-a-feature

Maybe we're not reading the same documentation?

Reply:
Ah, yeah, I was thinking of the current process which is framed around irreversible decisions. This is fine for reversible decisions.